controller/volume_controller.go

```diff
@@ -3049,7 +3195,7 @@ func (vc *VolumeController) IsReplicaUnavailable(r *longhorn.Replica) (bool, err
 		"replicaNodeID": r.Spec.NodeID,
 	})
 
-	if r.Spec.FailedAt != "" || r.Spec.NodeID == "" || r.Spec.DiskID == "" {
+	if r.Spec.NodeID == "" || r.Spec.DiskID == "" || r.Spec.DiskPath == "" || r.Spec.DataDirectoryName == "" {
```

Comment:
No, we don't need it. 
For failed migration replicas, the controller will add them into map `migrationReplicas` then do cleanup in function `deleteInvalidMigrationReplicas`. 
For failed old replicas, we can still clone migration replicas from them since this kind of duplicating is like replica reusage. The engine will determine if the cloned migration replicas are available based on the revision counter.